src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/DatabaseTreeNode.cs

```diff
@@ -102,5 +104,12 @@ private bool IsDWGen3(Database db)
                 && db.DatabaseEngineEdition == DatabaseEngineEdition.SqlDataWarehouse
                 && db.ServerVersion.Major == 12;
         }
+
+        private bool IsDataverse(Database db){
+            // Dataverses have a DatabaseEngineEdition of 1000. We have to cast it to int as DatabaseEngineEdition enum does not have a value for dataverse.
+            return db!= null
+                && (int)db.DatabaseEngineEdition == 1000 
```

Comment:
>nt)db.DatabaseEngineEdition == 1000  [](http://example.com/codeflow?start=21&length=36)

can we instead have a list of known valid editions and treat anything else the same way we treat datamart?

Reply:
you could use Enum.TryParse and if it doesn't have a value just assume it's data mart.

Reply:
Are you saying that datamart doesn't have a static engine edition? Why would we assume no mapped value == datamart otherwise? And what happens when a new edition is added later on - that could cause problems if we just assume it's data mart by default.

Reply:
My logic is this - we would add this value to the SMO enumeration only after we add real support for it. Once it shows up there presumably stuff would "just work" without the special cases in UI code.

Reply:
I'm saying that datamart is just one example of a TDS endpoint that isn't a real SQL Server. There could be others. The goal of the UI would be to provide a minimal OE for any such endpoint, correct?

Reply:
I could see that - where we basically say "if we don't know what type of server this is we default to some basic set of functionality that we know should work". So instead of having all this stuff be datamart-specific just have it check if it's a known edition and if not then skip certain logic. 

The SMO change to have an "Unknown" enum value would definitely help with that.